drivers/sensor/iis2mdc/iis2mdc.c

```diff
@@ -367,6 +315,78 @@ static int iis2mdc_init(const struct device *dev)
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, iis2mdc_init, device_pm_control_nop,
-		     &iis2mdc_data, &iis2mdc_dev_config, POST_KERNEL,
-		     CONFIG_SENSOR_INIT_PRIORITY, &iis2mdc_driver_api);
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "IIS2MDC driver enabled without any devices"
+#endif
+
+/*
+ * Device creation macro, shared by IIS2MDC_DEFINE_SPI() and
+ * IIS2MDC_DEFINE_I2C().
+ */
+
+#define IIS2MDC_DEVICE_INIT(inst)					\
+	DEVICE_DT_INST_DEFINE(inst,					\
+			    iis2mdc_init,				\
+			    device_pm_control_nop,			\
+			    &iis2mdc_data_##inst,			\
+			    &iis2mdc_config_##inst,			\
+			    POST_KERNEL,				\
+			    CONFIG_SENSOR_INIT_PRIORITY,		\
+			    &iis2mdc_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_IIS2MDC_TRIGGER
+#define IIS2MDC_CFG_IRQ(inst) \
+	.gpio_drdy = GPIO_DT_SPEC_GET(DT_DRV_INST(inst), drdy_gpios),
+#else
+#define IIS2MDC_CFG_IRQ(inst)
+#endif /* CONFIG_IIS2MDC_TRIGGER */
```

Comment:
Could simplify to

```
#define IIS2MDC_CFG_IRQ(inst) \
    IF_ENABLED(CONFIG_IIS2MDC_TRIGGER, (.gpio_drdy = GPIO_DT_SPEC_GET(DT_DRV_INST(inst), drdy_gpios),))
```

Not blocking on this one

Reply:
:(

Reply:
Sorry, I'm too tired. It's almost 10pm and I want to close this, as tomorrow I need to do other stuff. Sorry to turn this down. But I really appreciate your helps and guidance on this helpers usage!